plugins/vendor/zoom/ZoomDropFolderPlugin/batch/KZoomDropFolderEngine.php

```diff
@@ -0,0 +1,423 @@
+		{
+			$this->handleExistingDropFolderFile($dropFolderFile);
+		}
+		
+		if (self::$lastHandledMeetingTime)
+		{
+			self::updateDropFolderLastMeetingHandled(self::$lastHandledMeetingTime);
+		}
+	}
+	
+	protected function getMeetingsInStartTimeOrder()
+	{
+		$lastHandledDate = $this->dropFolder->lastHandledMeetingTime ? date('Y-m-d', $this->dropFolder->lastHandledMeetingTime) : 0;
+		$from = $lastHandledDate ? $lastHandledDate : date('Y-m-d',time() - self::MAX_DATE_RANGE_DAYS * self::ONE_DAY);
+		$to = date('Y-m-d', time());
+		$nextPageToken = '';
+		$pageSize = self::MAX_PAGE_SIZE;
+		$pageIndex = 0;
+		$meetingFilesByStartTime = array();
+		do
+		{
+			$resultZoomList = $this->zoomClient->listRecordings(self::ME, $from, $to, $nextPageToken, $pageSize);
+			$meetingFiles = $this->getMeetings($resultZoomList);
+			if (!$meetingFiles)
+			{
+				break;
+			}
+			foreach ($meetingFiles as $meetingFile)
+			{
+				$meetingsStartTime = self::convertTimeToUnix($meetingFile[self::START_TIME]);
+				$meetingFilesByStartTime[$meetingsStartTime] = $meetingFile;
+			}
+			
+			$pageIndex++;
+			$nextPageToken = $resultZoomList && $resultZoomList[self::NEXT_PAGE_TOKEN] ?
+				$resultZoomList[self::NEXT_PAGE_TOKEN] : '';
+			
+		} while ($nextPageToken !== '' && $pageIndex < 10);
+		
+		ksort($meetingFilesByStartTime);
+		return $meetingFilesByStartTime;
+	}
+	
+	protected function getMeetings($resultZoomList)
+	{
+		$meetings = $resultZoomList[self::MEETINGS];
+		if ($meetings)
+		{
+			KalturaLog::log('Found ['.count($meetings).'] in the folder');
+		}
+		else
+		{
+			KalturaLog::info('No physical files found for drop folder id ['.$this->dropFolder->id.']');
+			$meetings = array();
+		}
+		
+		KalturaLog::info("physical files: ");
+		foreach ($meetings as $meeting)
+		{
+			KalturaLog::info('Meeting UUID: '. $meeting[self::UUID]);
+		}
+		return $meetings;
+	}
+
+	protected function handleMeetingFiles($meetingFiles, &$dropFolderFilesMap)
+	{
+		foreach ($meetingFiles as $meetingFile)
+		{
+			KalturaLog::debug("meeting file is: " . print_r($meetingFile, true));
+			$recordingFilesOrdered = self::orderRecordingFiles($meetingFile[self::RECORDING_FILES]);
+			KalturaLog::debug("recording files are: " . print_r($recordingFilesOrdered, true));
+			foreach ($recordingFilesOrdered as $recordingFilesPerTimeSlot)
+			{
+				$firstDFFileOnTimeSlot = true;
+				$isParentEntry = false;
+				foreach ($recordingFilesPerTimeSlot as $recordingFile)
+				{
+					$recordingFileName = $meetingFile[self::UUID] . '_' . $recordingFile[self::ID] . ZoomHelper::SUFFIX_ZOOM;
+					if (!array_key_exists($recordingFileName, $dropFolderFilesMap))
+					{
+						if (ZoomHelper::shouldHandleFileType($recordingFile[self::RECORDING_FILE_TYPE]))
+						{
+							if ($firstDFFileOnTimeSlot)
+							{
+								$firstDFFileOnTimeSlot = false;
+								$entry = $this->getEntryByReferenceId(zoomProcessor::ZOOM_PREFIX . $meetingFile[self::UUID]);

```

Comment:
few issues - 
1. $entry is known in this scope, but used also in line 155 outside of the scope.
2. $isParentEntry - the name is confusing, its roles is more wasParentEntryFound

Instead of the 2 just use parameter - $parentEntry
If it is null, it was not found, if it has value - just use it.